app/frontend/src/components/ches/HistoryTable.vue

```diff
@@ -57,68 +59,113 @@
-      // TODO: implement promote and cancel actions
-      // {
-      //   text: 'Actions',
-      //   align: 'end',
-      //   value: 'actions',
-      //   filterable: false,
-      //   sortable: false,
-      // },
+      { text: 'Actions', align: 'start', value: 'actions', width: 120, filterable: false, sortable: false }
     ],
   }),
 
   computed: {
-    ...mapGetters('ches', ['txIds', 'tableData']),
+    ...mapGetters('ches', ['txIds']),
+    ...mapFields('ches', ['tableData', 'tableLoading'])
   },
 
   methods: {
     ...mapActions('alert', ['showAlert', 'clearAlert']),
-    ...mapActions('ches', ['addTableData', 'populateTable']),
+    ...mapActions('ches', ['populateTable', 'clearTable']),
+
 
-    cancelMessage(msgId) {
-      console.log('cancelMessage', msgId); // eslint-disable-line no-console
+    async cancelMessage(msgId) {
+      try {
+        // wait for OK status from CHES api
+        await chesService.cancel(msgId);
+        // update store with latest table data
+        this.populateTable();
+        // show success alert
+        this.showAlert({
+          type: 'success',
+          text: `<strong>Your message has been successfully cancelled.<br />Message ID: </strong> ${msgId}`,
+        });
+      } catch (e) {
+        this.error = true;
+        // show error alert
+        this.showAlert({
+          type: 'error',
+          text: e,
+        });
+      }
+      window.scrollTo(0, 0);
     },
 
     isTerminalState(status) {
       return ['completed', 'cancelled'].includes(status);
     },
 
-    promoteMessage(msgId) {
-      console.log('promoteMessage', msgId); // eslint-disable-line no-console
+    async promoteMessage(msgId) {
+      try {
+        // wait for OK status from CHES api (does not mean message has been promoted yet)
+        await chesService.promote(msgId);
+        // promote seems to take longer than cancel to complete and show in status from CHES
+        await new Promise(resolve => {
+          setTimeout(resolve, 1000);
+          this.tableLoading = true;
+        });
+        this.populateTable();
+        // show success alert
+        this.showAlert({
+          type: 'success',
+          text: `<strong>Your message has been successfully promoted.<br />Message ID: </strong> ${msgId}`,
+        });
+      } catch (e) {
+        this.error = true;
+        // show error alert
+        this.showAlert({
+          type: 'error',
+          text: e,
+        });
+      }
+      window.scrollTo(0, 0);
     },
 
-    humanDateTime(timestamp){
+    humanDateTime(timestamp) {
       // date-fns docs: https://date-fns.org/v2.21.1/docs/Getting-Started
       return timestamp ? format(new Date(timestamp), 'yyyy-MM-dd HH:mm') : '-';
-    }
+    },
   },
 
-  mounted() {
+  async mounted() {
     this.clearAlert();
     // get each message's status details in ches vuex module
-    this.populateTable();
+    await this.populateTable();
```

Comment:
This async/await wrapper is unnecessary - there's nothing following the mounted call that needs to explicitly block and wait on populateTable to complete.